### ISSUE TITLE
Coref demo produces bad results when input has consecutive whitespaces

### ISSUE THREAD

**Issue by ZhaofengWu** [allenai/allennlp#3779]
**Describe the bug**
When we give multiple sentences to the coreference demo where _consecutive_ whitespaces exist between sentences, the demo produces bad results with multiple nested co-referring spans. It's probably something related to the Spacy tokenizer that we're using internally.

**To Reproduce**
Steps to reproduce the behavior
1. Go to https://demo.allennlp.org/coreference-resolution/MTQyMDE2NA==
2. Choose the first sentence ("Paul Allen was..."); it parses nicely
3. Add one additional space after the first sentence (or alternatively copy-paste from below); it parses badly.

```
Paul Allen was born on January 21, 1953, in Seattle, Washington, to Kenneth Sam Allen and Edna Faye Allen.  Allen attended Lakeside School, a private school in Seattle, where he befriended Bill Gates, two years younger, with whom he shared an enthusiasm for computers. Paul and Bill used a teletype terminal at their high school, Lakeside, to develop their programming skills on several time-sharing computer systems.
```

**Comment by schmmd** [comment]
@ZhaofengWu would a reasonable fix be to simply replace multiple spaces with a single space when passing the input text to the model?

**Comment by ZhaofengWu** [comment]
I think that is a reasonable fix, but we should still investigate why we’re
seeing this.

**Comment by schmmd** [comment]
We need to understand whether this is a model issue or a demo issue.

**Comment by DeNeutoy** [comment]
Re-opening this, because we need to verify what the behaviour of the LM tokenizers is when they are passed whitespace tokens, see discussion in #3808

**Comment by ZhaofengWu** [comment]
Confirmed that this is due to the zero-length token produced by Huggingface transformer tokenizer, and was fixed by #3808
